analysis/rogueassassination/src/modules/shadowlands/legendaries/Doomblade.tsx

```diff
@@ -0,0 +1,70 @@
+import SPELLS from 'common/SPELLS';
+import Analyzer, { Options } from 'parser/core/Analyzer';
+import calculateEffectiveDamage from 'parser/core/calculateEffectiveDamage';
+import { SELECTED_PLAYER } from 'parser/core/EventFilter';
+import Events, { DamageEvent } from 'parser/core/Events';
+import Enemies from 'parser/shared/modules/Enemies';
+import BoringSpellValueText from 'parser/ui/BoringSpellValueText';
+import ItemDamageDone from 'parser/ui/ItemDamageDone';
+import Statistic from 'parser/ui/Statistic';
+import STATISTIC_CATEGORY from 'parser/ui/STATISTIC_CATEGORY';
+
+import { ASSASSINATION_BLEED_DEBUFFS } from '@wowanalyzer/rogue';
+
+class Doomblade extends Analyzer {
+  static dependencies = {
+    enemies: Enemies,
+  };
+  protected enemies!: Enemies;
+
+  bonusEnvenomDamage = 0;
+  mutilatedFleshDamage = 0;
+
+  constructor(options: Options) {
+    super(options);
+    this.active = this.selectedCombatant.hasLegendary(SPELLS.DOOMBLADE);
+    this.addEventListener(
+      Events.damage.by(SELECTED_PLAYER).spell(SPELLS.MUTILATED_FLESH),
+      this.onMutilatedFleshDamage,
+    );
+    this.addEventListener(
+      Events.damage.by(SELECTED_PLAYER).spell(SPELLS.ENVENOM),
+      this.onEnvenomDamage,
+    );
+  }
+
+  onMutilatedFleshDamage(event: DamageEvent) {
+    this.mutilatedFleshDamage += event.amount;
+  }
+
+  onEnvenomDamage(event: DamageEvent) {
+    const enemy = this.enemies.getEntity(event);
+    if (
+      enemy &&
+      ASSASSINATION_BLEED_DEBUFFS.some((effect) => enemy.hasBuff(effect.id, event.timestamp))
+    ) {
```

Comment:
not going to require it, but similar comment here as above: avoid having the entire method body wrapped in an if. you can just invert the if and return early. makes the code a little more manageable as it grows/changes.